--- ../../.share/scripts/load_icons.tcl	2014-01-05 14:23:32.000000000 -0200
+++ load_icons.tcl	2014-03-05 15:00:38.254441488 -0300
@@ -192,17 +192,17 @@
 				set fallback_off "led-off.png"
 			}
 			throttle {
-				set image_on     ""
-				set image_off    "${icon}.png"
-				set fallback_on  ""
-				set fallback_off ""
+				set image_on     "led-on.png"
+				set image_off    "led-off.png"
+				set fallback_on  "led-on.png"
+				set fallback_off "led-off.png"
 				set current_fade_delay_non_active($icon) 0
 			}
 			default {
-				set image_on     "${icon}.png"
-				set image_off    ""
-				set fallback_on  ""
-				set fallback_off ""
+				set image_on     "led-on.png"
+				set image_off    "led-off.png"
+				set fallback_on  "led-on.png"
+				set fallback_off "led-off.png"
 				set fade_delay_active($icon) 0
 			}
 		}
@@ -282,7 +282,7 @@
 
 # Available icons. Icons are also drawn in this order (by default)
 set icon_list [list "led_power" "led_caps" "led_kana" "led_pause" "led_turbo" "led_FDD" \
-                    "pause" "throttle" "mute" "breaked"]
+                    "pause" "throttle" "mute" "breaked" "code"]
 
 # create OSD widgets
 osd create rectangle osd_icons -scaled true -alpha 0 -z 1
